ios/Exponent/Kernel/Views/EXAppViewController.m

```diff
@@ -315,6 +318,34 @@ - (void)errorViewDidSelectRetry:(EXErrorView *)errorView
 
 - (UIInterfaceOrientationMask)supportedInterfaceOrientations
 {
+  NSString *validatedVersion = [[EXVersions sharedInstance] availableSdkVersionForManifest:_appRecord.appLoader.manifest];
+  NSString *majorSDKVersion = [validatedVersion componentsSeparatedByString:@"."][0];
```

Comment:
This method in general looks like this:
```
if SDK <= 35
  use legacySupportedInterfaceOrientations
if we have a mask in the singleton module
  use that mask
use legacySupportedInterfaceOrientations (and save calculated mask to singleton)
```

couldn't we simplify the method to

```
if we have a mask in the singleton module
  use that mask
use legacySupportedInterfaceOrientations
```
?

This way the method becomes much shorter, we explicitly reuse `legacySupportedInterfaceOrientations`, and if we know that if there is a mask for a given experience in the registry — it has been set by the JS and it overrides everything else